src/server/service/match.ts

```diff
@@ -276,6 +276,10 @@ export class Match {
         (p) => p.userId !== player.userId,
       );
 
+      // Checking if the player did send a message this match
+      if (!this._messages.some((message) => message.sender === player.userId))
+        return { ...player };
```

Comment:
You can just check if player voted. this is redundant and out of the scope of this method